distributed/tests/test_worker.py

```diff
@@ -407,6 +407,29 @@ def __str__(self):
         assert "Bar" in str(e.__cause__)
 
 
+@pytest.mark.asyncio
+async def test_plugin_exception():
+    class MyException(Exception):
+        def __init__(self, msg):
+            self.msg = msg
+
+        def __str__(self):
+            return "MyException(%s)" % self.msg
+
+    class MyPlugin:
+        def setup(self, worker=None):
+            raise MyException("Foo")
+
+    s = await Scheduler(port=8007)
```

Comment:
Generally we try to create `Scheduler` instance with a context manager in tests to ensure things are cleaned up properly.

```python
async with Scheduler() as s:
```

Also, is there a reason we need to specify `port=8007` here?

Reply:
Oops, I see @mrocklin made a similar comment. Marking this one as resolved